warning[W09002]: unused variable
   ┌─ tests/linter/move_2024/unneeded_return_okay.move:14:9
   │
14 │     let y = &mut x;
   │         ^ Unused local variable 'y'. Consider removing or prefixing with an underscore: '_y'
   │
   = This warning can be suppressed with '#[allow(unused_variable)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

error[E04007]: incompatible types
   ┌─ tests/linter/move_2024/unneeded_return_okay.move:15:14
   │
13 │     let mut x = 0;
   │         ----- Given: integer
14 │     let y = &mut x;
15 │     return (*x = 1)
   │              ^
   │              │
   │              Invalid mutation. Expected a mutable reference
   │              Expected: '&mut _'

error[E04007]: incompatible types
   ┌─ tests/linter/move_2024/unneeded_return_okay.move:19:12
   │
19 │ fun t3() { 'a: { return { return 'a 0 } } }
   │     --     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   │     │      │
   │     │      Invalid return expression
   │     │      Given: integer
   │     Expected: '()'

error[E04014]: invalid loop control
   ┌─ tests/linter/move_2024/unneeded_return_okay.move:23:19
   │
23 │ fun t4() { return continue }
   │                   ^^^^^^^^ Invalid usage of 'continue'. 'continue' can only be used inside a loop body or lambda

error[E04014]: invalid loop control
   ┌─ tests/linter/move_2024/unneeded_return_okay.move:25:19
   │
25 │ fun t5() { return break }
   │                   ^^^^^ Invalid usage of 'break'. 'break' can only be used inside a loop body or lambda

error[E04014]: invalid loop control
   ┌─ tests/linter/move_2024/unneeded_return_okay.move:27:20
   │
27 │ fun t6() { return (break + 5) }
   │                    ^^^^^ Invalid usage of 'break'. 'break' can only be used inside a loop body or lambda

error[E03005]: unbound unscoped name
   ┌─ tests/linter/move_2024/unneeded_return_okay.move:29:19
   │
29 │ fun t7() { return unnamed_fun() }
   │                   ^^^^^^^^^^^ Unbound function 'unnamed_fun' in current scope

